python/lsst/cell_coadds/_grid_container.py

```diff
@@ -0,0 +1,201 @@
+# This file is part of cell_coadds.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+from collections.abc import MutableMapping
+from itertools import product
+from typing import TYPE_CHECKING, Callable, Iterable, Iterator, TypeVar
+
+from lsst.skymap import Index2D
+
+from ._uniform_grid import UniformGrid
+
+if TYPE_CHECKING:
+    import lsst.geom as geom
+
+
+__all__ = ("GridContainer",)
+
+T = TypeVar("T")
+
+
+class GridContainer(MutableMapping[Index2D, T]):  # Generic, Collection
```

Comment:
I realized that I could just use the type hints for `MutableMapping` and inherit from it, instead of trying the `Generic` or `Collection` types.